src/JMS/Serializer/Metadata/Driver/DoctrinePHPCRTypeDriver.php

```diff
@@ -64,6 +67,18 @@ protected function setPropertyType(DoctrineClassMetadata $doctrineMetadata, Prop
             }
 
             $propertyMetadata->setType($targetEntity);
+            return;
+        }
+
+        switch ($propertyName) {
+            case $doctrineMetadata->nodename:
```

Comment:
Don't you have to check before that `$doctrineMetadata` actually is an instance of `Doctrine\ODM\PHPCR\Mapping\ClassMetadata`?